modules/sievefilters/hm-sieve-pool.php

```diff
@@ -0,0 +1,144 @@
+<?php
+ini_set('memory_limit', '2048M');
+
+use PhpSieveManager\ManageSieve\Client;
+class SieveConnectionPool
+{
+    private static $config = [];
+    private static $connections = [];
+    private static $lastConnectionTimes = [];
+    /**
+     * Default timeout for connections in seconds.
+     */
+    private static $timeout = 600;
+
+    /**
+     * Cache TTL in seconds
+     */
+    private static $scriptCacheTTL = 600;
+
+    /**
+     * Cache instance (injected or global)
+     */
+    private static $cache = null;
+
+    private function __construct() {}
+
+    /**
+     * Optionally inject a global cache instance (Hm_Cache)
+     */
+    public static function setCache($cacheInstance)
+    {
+        if(!self::$cache) {
+            self::$cache = $cacheInstance;
+        }
+    }
+
+    /**
+     * Set the configuration for all servers
+     */
+    public static function setConfig(array $servers)
+    {
+        if (empty($servers)) {
+            throw new Exception("Configuration cannot be empty.");
+        }
+        foreach ($servers as $key => $server) {
+            if (!isset($server['host'], $server['port'], $server['username'], $server['password'])) {
+                throw new Exception("Each server configuration must include 'host', 'port', 'username', and 'password'.");
+            }
+            if (!is_string($server['host']) || !is_int($server['port']) || !is_string($server['username']) || !is_string($server['password'])) {
+                throw new Exception("Invalid types in server configuration for '$key'.");
+            }
+            if (isset(self::$config[$key])) {
+                self::$config[$key] = array_merge(self::$config[$key], $server);
+            } else {
+                self::$config[$key] = $server;
+            }
+        }
+    }
+
+    /**
+     * Get a Sieve connection by its key (server identifier)
+     */
+    public static function get($key)
+    {
+        if (!isset(self::$config[$key])) {
+            throw new Exception("No configuration found for '$key'");
+        }
+
+        if (!isset(self::$connections[$key]) || !self::isAlive($key)) {
+            self::$connections[$key] = self::connectServer(self::$config[$key]);
+            self::$lastConnectionTimes[$key] = time();
+        }
+
+        return self::$connections[$key];
```

Comment:
If you want to implement a connection reuse for a single web request, you can just add a global connection singleton or facade that returns an existing connection if available, otherwise, it starts a new connection and stores it globally, so it is accessible for code executed down the road.

If you want to implement a connection pool that can be shared between requests that's a totally different story and the approach should be different. I am not sure we can cache/serialize an active connection, so it can be reused in a later ajax or normal request - you can test that but if it is possible, then you can actually implement a connection pool, spin connections when necessary and return unserialized active connections when available.